src/priority_fee.rs

```diff
@@ -193,58 +196,120 @@ impl GrpcConsumer for PriorityFeeTracker {
 
 impl PriorityFeeTracker {
     pub fn new(slot_cache_length: usize) -> Self {
+        let (sampling_txn, sampling_rxn) = channel::<(Vec<Pubkey>, bool, Option<u32>)>(1_000);
+
         let tracker = Self {
             priority_fees: Arc::new(DashMap::new()),
             slot_cache: SlotCache::new(slot_cache_length),
             compute_budget: ComputeBudget::default(),
+            sampling_sender: sampling_txn,
         };
-        tracker.poll_fees();
+        tracker.poll_fees(sampling_rxn);
         tracker
     }
 
-    fn poll_fees(&self) {
+    fn poll_fees(&self, mut sampling_rxn: Receiver<(Vec<Pubkey>, bool, Option<u32>)>) {
```

Comment:
The thread will be blocked on sampling queue waiting for any new sample to compare both algos to. If nothing is in queue for 1 second than the global sample will be taken and logged to statsd